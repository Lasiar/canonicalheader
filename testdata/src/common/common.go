package common

import (
	"net/http"
)

const constTestHeader = "testHeaderValue"

func p() {
	v := http.Header{}
	v.Get(constTestHeader) // want `use "Testheadervalue" instead of "testHeaderValue"`

	v.Get("Test-HEader")           // want `use "Test-Header" instead of "Test-HEader"`
	v.Set("Test-HEader", "value")  // want `use "Test-Header" instead of "Test-HEader"`
	v.Add("Test-HEader", "value")  // want `use "Test-Header" instead of "Test-HEader"`
	v.Del("Test-HEader")           // want `use "Test-Header" instead of "Test-HEader"`
	v.Values("Test-HEader")        // want `use "Test-Header" instead of "Test-HEader"`
	v.Values(`Raw-STRING-Literal`) // want `use "Raw-String-Literal" instead of "Raw-STRING-Literal"`

	v.Set("Test-Header", "value")
	v.Add("Test-Header", "value")
	v.Del("Test-Header")
	v.Values("Test-Header")

	var someString = ""
	v.Get(someString)

	v.Write(nil)
	v.Clone()
	v.WriteSubset(nil, nil)
}
