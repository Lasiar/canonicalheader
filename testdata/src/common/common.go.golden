package common

import (
	"net/http"
)

const constTestHeader = "testHeaderValue"

func p() {
	v := http.Header{}
	v.Get(constTestHeader) // want `use "Testheadervalue" instead of "testHeaderValue"`

	v.Get("Test-Header")           // want `use "Test-Header" instead of "Test-HEader"`
	v.Set("Test-Header", "value")  // want `use "Test-Header" instead of "Test-HEader"`
	v.Add("Test-Header", "value")  // want `use "Test-Header" instead of "Test-HEader"`
	v.Del("Test-Header")           // want `use "Test-Header" instead of "Test-HEader"`
	v.Values("Test-Header")        // want `use "Test-Header" instead of "Test-HEader"`
	v.Values(`Raw-String-Literal`) // want `use "Raw-String-Literal" instead of "Raw-STRING-Literal"`

	v.Set("Test-Header", "value")
	v.Add("Test-Header", "value")
	v.Del("Test-Header")
	v.Values("Test-Header")

	var someString = ""
	v.Get(someString)

	v.Write(nil)
	v.Clone()
	v.WriteSubset(nil, nil)
}
